File: jeongsiyeong/python/codetree/binarySearch_03.py
N, M = map(int, input().split())

arr = list(map(int, input().split()))

querys = list(map(int, input().split()))

def bin_search(q):
    left = 0
    right = N - 1
    answer = -1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == q:
            answer = mid+1
            right = mid - 1
        elif arr[mid] < q :
            left = mid + 1
        else:
            right = mid - 1
    return answer

for query in querys:
    print(bin_search(query))

File: jeongsiyeong/cpp/codetree/binarySearch_03.cpp
#include <iostream>
#include <vector>

using namespace std;

int n,m;
vector<int> arr;
vector<int> querys;

int bin_search(int q){
  int left = 0;
  int right = n -1;
  int answer = -1;

  while(left <= right){
    int mid = left + (right - left ) / 2;

    if (arr[mid] == q){
      answer = mid + 1;
      right = mid - 1;
    }else if (arr[mid] < q){
      left = mid + 1;
    }else{
      right = mid - 1;
    }
  }
  return answer;
}

int main(){
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);

  arr.resize(n);
  for(int i=0; i<n; i++){
    cin >> arr[i];
  }

  querys.resize(m);
  for(int i=0; i<m; i++){
    cin >> querys[i];
  }

  for(int i=0; i<m; i++){
    cout<<bin_search(querys[i]) <<"\n";
  }
}